error: unnecessary qualification
  --> dexios-domain/src/key/add.rs:30:23
   |
30 |     let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> dexios-domain/src/lib.rs:42:5
   |
42 |     unused_qualifications
   |     ^^^^^^^^^^^^^^^^^^^^^
help: remove the unnecessary path segments
   |
30 -     let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
30 +     let (header, _) = Header::deserialize(&mut *req.handle.borrow_mut())
   |

error: unnecessary qualification
  --> dexios-domain/src/key/change.rs:30:23
   |
30 |     let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the unnecessary path segments
   |
30 -     let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
30 +     let (header, _) = Header::deserialize(&mut *req.handle.borrow_mut())
   |

error: unnecessary qualification
  --> dexios-domain/src/key/delete.rs:22:23
   |
22 |     let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the unnecessary path segments
   |
22 -     let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
22 +     let (header, _) = Header::deserialize(&mut *req.handle.borrow_mut())
   |

error: could not compile `dexios-domain` (lib) due to 3 previous errors

---

--- a/dexios-domain/src/key/add.rs
+++ b/dexios-domain/src/key/add.rs
@@ -27,7 +27,7 @@ pub fn execute<RW>(req: Request<'_, RW>)
 where
     RW: Read + Write + Seek,
 {
-    let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
+    let (header, _) = Header::deserialize(&mut *req.handle.borrow_mut())
         .map_err(|_| Error::HeaderDeserialize)?;
 
     if header.header_type.version < HeaderVersion::V5 {
--- a/dexios-domain/src/key/change.rs
+++ b/dexios-domain/src/key/change.rs
@@ -27,7 +27,7 @@ pub fn execute<RW>(req: Request<'_, RW>)
 where
     RW: Read + Write + Seek,
 {
-    let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
+    let (header, _) = Header::deserialize(&mut *req.handle.borrow_mut())
         .map_err(|_| Error::HeaderDeserialize)?;
 
     if header.header_type.version < HeaderVersion::V5 {
--- a/dexios-domain/src/key/delete.rs
+++ b/dexios-domain/src/key/delete.rs
@@ -19,7 +19,7 @@ pub fn execute<RW>(req: Request<'_, RW>)
 where
     RW: Read + Write + Seek,
 {
-    let (header, _) = core::header::Header::deserialize(&mut *req.handle.borrow_mut())
+    let (header, _) = Header::deserialize(&mut *req.handle.borrow_mut())
         .map_err(|_| Error::HeaderDeserialize)?;
 
     if header.header_type.version < HeaderVersion::V5 {
